java/util/HashMap/HashMapTest.cpp

```diff
@@ -33,444 +33,835 @@ using namespace Java::Lang;
+    HashMap<String, String> anotherMap = hashMap.clone();
+
+    // Test actualKey = "Key1" is exist in anotherMap
+    actualKey = "Key1";
+    expected = "Value of Key1";
+    actual = anotherMap.get(actualKey);
+    assertEquals(expected.toString(), actual.toString());
+
+    // Test actualKey = "Key2" is exist in anotherMap
+    actualKey = "Key2";
+    expected = "Value of Key2";
+    actual = anotherMap.get(actualKey);
+    assertEquals(expected.toString(), actual.toString());
+
+    // Test non-exist key
+    actualKey = "wrong key";
+    actual = anotherMap.get(actualKey);
+    assertTrue(actual.isEmpty());
 }
 
 TEST (JavaUtil, HashMapContainsKey) {
-	// Given valid hash map to test containsKey()
-	HashMap<Integer, String> hashMap;
-	hashMap.put(3, "three");
-	hashMap.put(17, "seven teen");
-	hashMap.put(-52, "negative fifty two");
+    boolean actual;
+
+    // Create a HashMap to test
+    HashMap<Integer, String> hashMap;
+    hashMap.put(3, "three");
+    hashMap.put(17, "seven teen");
+    hashMap.put(-52, "negative fifty two");
 
-	boolean exist = hashMap.containsKey(-52);
-	assertTrue(exist);
+    // Valid key
+    actual = hashMap.containsKey(-52);
+    assertTrue(actual);
 
-	boolean notFound = hashMap.containsKey(100);
-	assertFalse(notFound);
+    // Invalid key
+    actual = hashMap.containsKey(100);
+    assertFalse(actual);
 }
 
 TEST (JavaUtil, HashMapContainsValue) {
-	// Given valid hash map to test containsValue()
-	HashMap<String, Double> hashMap;
-	hashMap.put("15.3", 15.3);
-	hashMap.put("30.111", 30.111);
+    boolean actual;
 
-	boolean exist = hashMap.containsValue(15.3);
-	assertTrue(exist);
+    // Create a HashMap to test
+    HashMap<String, Double> hashMap;
+    hashMap.put("15.3", 15.3);
+    hashMap.put("30.111", 30.111);
 
-	boolean notFound = hashMap.containsValue(30.22);
-	assertFalse(notFound);
+    // Valid key
+    actual = hashMap.containsValue(15.3);
+    assertTrue(actual);
+
+    // Invalid key
+    actual = hashMap.containsValue(30.22);
+    assertFalse(actual);
 }
 
 TEST (JavaUtil, HashMapEntrySet) {
-	HashMap<String, String> hashMap;
-	for (int index=1; index<=100; index++) {
-		hashMap.put("Key "+ String::valueOf(index), "Value " + String::valueOf(index));
-	}
-	int counter = 0;
-	Set<class Map<String, String>::Entry> entrySet = hashMap.entrySet();
-	// TODO - loint@foodtiny.com will improve entrySet
-	// then we can put it inside foreach without any performance issue
-	for (Map<String, String>::Entry entry : entrySet) {
-		counter += 1;
-		if (counter == 1) {
-			assertEquals("Key 99", entry.getKey().toString());
-			assertEquals("Value 99", entry.getValue().toString());
-		}
-		if (counter == 2) {
-			assertEquals("Key 98", entry.getKey().toString());
-			assertEquals("Value 98", entry.getValue().toString());
-		}
-	}
-	// Make sure foreach is working
-	assertEquals(100, counter);
+    HashMap<String, String> hashMap;
+
+    for (int index = 1; index <= 100; index++) {
```

Comment:
Please create `int index;` outside the loop

Reply:
Thanks. Fixed.